presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -619,8 +628,19 @@ private synchronized void sendUpdate()
                 outputBuffers.get(),
                 totalPartitions);
         byte[] taskUpdateRequestJson = taskUpdateRequestCodec.toBytes(updateRequest);
+
+        if (taskUpdateRequestJson.length >= taskUpdateSizeLimit) {
+            throw new PrestoException(EXCEEDED_TASK_UPDATE_SIZE_LIMIT, format("TaskUpdate size exceeded the limit of %d", taskUpdateSizeLimit));
+        }
+
         if (fragment.isPresent()) {
-            stats.updateWithPlanBytes(taskUpdateRequestJson.length);
+            stats.updateWithPlanSize(taskUpdateRequestJson.length);
+        }
+        else {
+            // This is to keep track of the task update size even when the plan fragment is NOT present
+            if (ThreadLocalRandom.current().nextInt(100) == 1) {
```

Comment:
Make this into a separate commit. 

Also I set a private final static double value `UPDATE_WITHOUT_PLAN_STATS_SAMPLE_RATE = 0.01`, and just use `ThreadLocalRandom.current().nextDouble() < UPDATE_WITHOUT_PLAN_TRACKING_SAMPLE_RATE`